sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -653,6 +768,71 @@ def test_gaussian_mixture_fit():
             assert_allclose(ecov.error_norm(prec_pred[k]), 0, atol=0.15)
 
 
+def test_weighted_vs_repeated():
+    # a sample weight of N should yield the same result as an N-fold
+    # repetition of the sample
+    rng = np.random.RandomState(0)
+    n_samples = 200
+    rand_data = RandomData(rng, n_samples=n_samples)
+    n_repeated_samples = 80
+
+    n_features = rand_data.n_features
+    n_components = rand_data.n_components
+
+    for covar_type in COVARIANCE_TYPE:
+        X = rand_data.X[covar_type]
+
+        sample_weight = np.ones(n_samples)
+        sample_weight[n_repeated_samples:] += 2
+
+        X_second_component = X[n_repeated_samples:]
+        X_repeat = np.vstack((X, X_second_component, X_second_component))
+
+        g = GaussianMixture(n_components=n_components, n_init=20,
+                            reg_covar=0, random_state=rng,
+                            covariance_type=covar_type)
+
+        g1 = GaussianMixture(n_components=n_components, n_init=20,
+                             reg_covar=0, random_state=rng,
+                             covariance_type=covar_type)
+
+        g_weighted = g.fit(X, sample_weight=sample_weight)
+        g_repeated = g1.fit(X_repeat)
+
+        assert_allclose(np.sort(g_weighted.weights_),
+                        np.sort(g_repeated.weights_))
+
+        arg_idx1 = g_weighted.means_[:, 0].argsort()
+        arg_idx2 = g_repeated.means_[:, 0].argsort()
+        assert_allclose(g_weighted.means_[arg_idx1],
+                        g_repeated.means_[arg_idx2])
+
+        if covar_type == 'full':
```

Comment:
I can't say we'd always go to this level of detail in ensuring models are identical.

It would be less repetitive if you defined the transformation once, then applied it to both `precisions_` attributes.

```py
def uniform_precisions(precisions):
    if covar_type == "tied":
        ...
    ...
    return np.trace(precisions, axis1=1, axis2=2).argsort()
```

But I also wonder whether we should just more approximately evaluate that the two models have identical likelihood functions applied to held-out data.